packages/core/types/src/types/core/attributes/component.ts

```diff
@@ -27,7 +27,16 @@ export type ComponentValue<
   TComponentUID extends Common.UID.Component,
   TRepeatable extends Utils.Expression.BooleanValue
 > = Attribute.GetValues<TComponentUID> extends infer TValues
-  ? Utils.Expression.If<TRepeatable, TValues[], TValues>
+  ? Utils.Expression.MatchFirst<
+      [
+        // Repeatable component
+        [Utils.Expression.IsTrue<TRepeatable>, TValues[]],
+        // Single component
+        [Utils.Expression.IsFalse<TRepeatable>, TValues]
+      ],
+      // If TRepeatable is neither true nor false, then return every possible types
+      TValues[] | TValues
```

Comment:
🤔  in the case the repeatable field is not set in the schema we still set it to false so can we make it required so we don't need that fallback that doesn't mean anything ?

Reply:
Here it's more about "I don't know wether it's false or true, not if this is undefined

Reply:
In short: `Attribute.Component` is a component that can be repeatable or not (`true | false` -> `boolean`), by making the check on `.repeatable`, you should be able to infer the correct values

Reply:
but this should always be one or the other (we are missing a `is` operator somewhere :D) but in that case I would return a never instead to represent there is not valid return type 🤷